src/Functions/URL/domain.h

```diff
@@ -13,6 +13,9 @@ namespace
 
 inline StringRef checkAndReturnHost(const Pos & pos, const Pos & dot_pos, const Pos & start_of_host)
 {
+    if (!dot_pos && *pos == ':' && pos > start_of_host)
```

Comment:
This change affects some other URL functions, and the biggest problem here is `select port('http:80/paul/example/com')` ->  `80`, while this should return 0 since the URL is invalid.

`port` function works as expected for URLs (and it is documented to accept URL), but not for hosts, so maybe introduce a separate function, i.e. `hostPort`?

Reply:
Maybe you should add an example showing your concerns to the test

Reply:
Does `https://clickhouse:80` can be a valid URL? Or we just need to consider `localhost` hostname?

Reply:
I mean this one: `http:80/paul/example/com`

Reply:
For `https://localhost:80/`, in original code, get host will return empty string.

Reply:
Also consider

`https://user:password@[2a02:6b8:1234:abcd::1:1]:port/hello:world?abc:def=g:h#test:test`